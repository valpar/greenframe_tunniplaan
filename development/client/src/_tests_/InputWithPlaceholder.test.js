import React from 'react';
import { render, fireEvent, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import InputWithPlaceholder from '../components/UI/Input/InputWithPlaceholder.js';


describe('InputWithPlaceholder', () => {
  it("matches snapshot", () => {
    const tree = renderer.create(<InputWithPlaceholder />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it("renders with provided placeholder and type", () => {
    render(<InputWithPlaceholder placeholder="Test Placeholder" name="testInput" type="email" />);
    expect(screen.getByRole('textbox')).toHaveAttribute('type', 'email');
  });
  it('calls onChange on input change', () => {
    const handleChange = jest.fn();
    render(<InputWithPlaceholder onChange={handleChange} />);
    fireEvent.change(screen.getByRole('textbox'), { target: { value: 'new value' } });
    expect(handleChange).toHaveBeenCalledWith({
      name: undefined,
      value: 'new value',
    });
  });
  it('shows tooltip on mouse enter when error message is present', () => {
    render(<InputWithPlaceholder errorMessage="Error" />);
    const input = screen.getByRole('textbox');
    fireEvent.mouseEnter(input);
    expect(screen.getByText("Error")).toBeInTheDocument();
  });
  it('hides tooltip on mouse leave', () => {
    render(<InputWithPlaceholder errorMessage="Error" />);
    const input = screen.getByRole('textbox');
    fireEvent.mouseEnter(input);
    fireEvent.mouseLeave(input);
    expect(screen.queryByText("Error")).not.toBeInTheDocument();
  });
});