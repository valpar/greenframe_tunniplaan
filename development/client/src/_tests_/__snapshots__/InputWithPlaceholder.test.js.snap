// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputWithPlaceholder matches snapshot 1`] = `
<div
  className="relative flex flex-col w-full"
>
  <input
    autoComplete="off"
    className="p-2 border border-borderGray  rounded-none font-normal focus:outline-none focus:border-black"
    name=""
    onBlur={[Function]}
    onChange={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    readOnly={false}
    type="text"
    value=""
  />
</div>
`;
